internal/validation/v2alpha1/rules.go

```diff
@@ -56,6 +58,24 @@ func validateRules(ctx context.Context, client client.Client, parentAttributePat
 	return problems
 }
 
+func validatePath(validationPath string, rulePath string) (problems []validation.Failure) {
+	problems = append(problems, validateEnvoyTemplate(validationPath+".path", rulePath)...)
+	return problems
+}
+
+func validateEnvoyTemplate(s string, path string) []validation.Failure {
+	segments := strings.Split(strings.TrimLeft(path, "/"), "/")
+	doubleAsteriskExists := false
+	for _, segment := range segments {
+		if segment == "{**}" && !doubleAsteriskExists {
+			doubleAsteriskExists = true
+		} else if (segment == "{*}" || segment == "{**}") && doubleAsteriskExists {
+			return []validation.Failure{{AttributePath: s, Message: fmt.Sprintf("Operator %s was used after operator {**}. The {**} operator must be the last in the path.", segment)}}
```

Comment:
Problem is that in case you do not return a slice, you must always return value. This does not work with how we handle the failures, as the slice length will increase

https://go.dev/play/p/VMYDiP3KY8f

Reply:
But you can just return a single `validation.Failure` and append only one element in L62
```
problems = append(problems, validatePath(ruleAttributePath, rule.Path))
```